### PR TITLE
Deprecate `npartitions='auto'` for set_index & sort_values

### LINKED ISSUE #10736: Deprecate ``npartitions="auto"`` for set_index and sort_values
The docs state that this makes a decision based on memory use

> If ‘auto’ then decide by memory use.

But that's just wrong

```
    if npartitions == "auto":
        repartition = True
        npartitions = max(100, df.npartitions)
```

This is very prohibitive for larger workloads, imagine a DataFrame with thousands of partitions getting compressed that heavily, so let's just get rid of this option

### PR DESCRIPTION
**Pull request by milesgranger**
- [x] Closes #10736 
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

File: dask/dataframe/tests/test_shuffle.py [hunk 1]
```diff
@@ -569,19 +569,19 @@ def _set_index(i, df, idx):
     return df.set_index(idx).divisions
 
 
-def test_set_index_reduces_partitions_small(shuffle_method):
+def test_set_index_not_reduce_partitions_small(shuffle_method):
```
Comment (phofl) [inline dask/dataframe/tests/test_shuffle.py:572]: I think you can just rip out this test

File: dask/dataframe/shuffle.py [hunk 2]
```diff
@@ -166,6 +166,7 @@ def sort_values(
         return df.map_partitions(sort_function, **sort_kwargs)
 
     if npartitions == "auto":
+        warnings.warn("`npartitions='auto'` is deprecated.", FutureWarning, 2)
```
Comment (phofl) [inline dask/dataframe/shuffle.py:169]: Can you say something what users should do instead

File: dask/dataframe/tests/test_shuffle.py [hunk 3]
```diff
@@ -569,19 +569,19 @@ def _set_index(i, df, idx):
...
+    ddf2 = ddf.set_index("x", shuffle_method=shuffle_method)
+    assert ddf2.npartitions == ddf.npartitions
 
 
 def make_part(n):
     return pd.DataFrame({"x": np.random.random(n), "y": np.random.random(n)})
 
 
-def test_set_index_reduces_partitions_large(shuffle_method):
+def test_set_index_not_reduce_partitions_large(shuffle_method):
```
Comment (phofl) [inline dask/dataframe/tests/test_shuffle.py:584]: Same here, the test doesn't make much sensee anymore

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**